### ISSUE TITLE
Add new formatter for GitHub Actions

### ISSUE THREAD

**Issue by ybiquitous** [stylelint/stylelint#4698]
> What is the problem you're trying to solve?

When running stylelint on GitHub Actions, I'd like to see annotations of stylelint violations. For example:

<img width="998" alt="image" src="https://user-images.githubusercontent.com/473530/78919415-d72a1d80-7acc-11ea-82bd-6c65e921d8d4.png">

> What solution would you like to see?

A new formatter implemented by using [Actions workflow commands](https://help.github.com/en/actions/reference/workflow-commands-for-github-actions). Here is a PoC:

https://github.com/ybiquitous/stylelint-github-actions-formatter/pull/1/files

- For warning: `::warning file={name},line={line},col={col}::{message}`
- For error: `::error file={name},line={line},col={col}::{message}`

---

Note: ESLint's [`stylish`](https://github.com/eslint/eslint/blob/54630f0063fa66318476473757b5912465d4caf2/lib/cli-engine/formatters/stylish.js) formatter has already realized the feature, but it does **not** seem that it uses the Actions workflow commands. I don't know how ESLint realizes it. 🤷‍♂️ 
(Perhaps there may be a better solution than adding a new formatter.)

**Comment by ybiquitous** [comment]
I understand how ESLint works on Actions without any configuration: **Problem Matchers**
https://github.com/actions/toolkit/blob/master/docs/problem-matchers.md

Also, the following pull request has been opened:
https://github.com/actions/setup-node/pull/98

We may not need to address this issue.

**Comment by jeddy3** [comment]
This would be great to have. Let's hope it can be resolved upstream in `setup-node`. If not, we should rustle up a solution ourselves.

**Comment by ybiquitous** [comment]
Here is a simple workaround without adding a new formatter:

```yaml
name: stylelint

on: pull_request

jobs:
  check:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - run: npm install stylelint stylelint-config-standard
      - run: |
          echo '{"extends": "stylelint-config-standard"}' >> .stylelintrc.json
      - run: |
          npx stylelint --formatter unix '**/*.{css,scss}' \
            | ruby -pe 'sub(/^(.+):(\d+):(\d+): (.+) \[(.+)\]$/, %q{::\5 file=\1,line=\2,col=\3::\4})'
        shell: bash
```

See also:
- https://gist.github.com/ybiquitous/184a337662bfcafdd3d227253f9401ff
- https://github.com/ybiquitous/stylelint-github-actions-formatter/pull/2

**Comment by skhaz** [comment]
+1

**Comment by hazem454** [comment]
> > What is the problem you're trying to solve?
> 
> 
> 
> When running stylelint on GitHub Actions, I'd like to see annotations of stylelint violations. For example:
> 
> 
> 
> <img width="998" alt="image" src="https://user-images.githubusercontent.com/473530/78919415-d72a1d80-7acc-11ea-82bd-6c65e921d8d4.png">
> 
> 
> 
> > What solution would you like to see?
> 
> 
> 
> A new formatter implemented by using [Actions workflow commands](https://help.github.com/en/actions/reference/workflow-commands-for-github-actions). Here is a PoC:
> 
> 
> 
> https://github.com/ybiquitous/stylelint-github-actions-formatter/pull/1/files
> 
> 
> 
> - For warning: `::warning file={name},line={line},col={col}::{message}`
> 
> - For error: `::error file={name},line={line},col={col}::{message}`
> 
> 
> 
> ---
> 
> 
> 
> Note: ESLint's [`stylish`](https://github.com/eslint/eslint/blob/54630f0063fa66318476473757b5912465d4caf2/lib/cli-engine/formatters/stylish.js) formatter has already realized the feature, but it does **not** seem that it uses the Actions workflow commands. I don't know how ESLint realizes it. 🤷‍♂️ 
> 
> (Perhaps there may be a better solution than adding a new formatter.)
> 
>
